proto-build/src/main.rs

```diff
@@ -379,6 +379,17 @@ fn copy_and_patch(src: impl AsRef<Path>, dest: impl AsRef<Path>) -> io::Result<(
         // Use `tendermint_proto` as source of `google.protobuf` types
         // TODO(tarcieri): figure out what's wrong with our `buf` config and do it there
         ("::prost_types::", "::tendermint_proto::google::protobuf::"),
+        // add the feature flag to the serde definitions
+        (
+            "impl serde::Serialize for",
+            "#[cfg(feature = \"serde\")]\n\
+            impl serde::Serialize for",
+        ),
+        (
+            "impl<'de> serde::Deserialize<'de> for",
+            "#[cfg(feature = \"serde\")]\n\
+            impl<'de> serde::Deserialize<'de> for",
+        ),
```

Comment:
These can probably be gated on the module level but I don't see any reason to block the PR over that